rfd/0093-offline-access.md

```diff
@@ -0,0 +1,216 @@
+
+However, the decision if the connection should be established is ultimately made by the node, not the Auth server. The node
+performs an RBAC check on all inbound connections to determines if among other things per-session MFA is required and the MFA ceremony was 
+performed. Thus, `tsh` should **always** attempt to connect to a node even if the `proto.AuthService/IsMFARequired` RPC fails. In the worst
+case this makes `tsh ssh` for per-session MFA enabled connections take longer to fail. But that cost comes at the ability to allow all
+users that will never have to perform per-session MFA to be able to connect to nodes without needing the Auth server if they already 
+have certificates.
+
+Successful connection attempt when Auth isn't available and per-session MFA is **not** required
+```mermaid
+sequenceDiagram
+    participant tsh
+    participant Auth
+    participant Node
+
+    tsh->>+Auth: IsMFARequired
+    Auth-->>-tsh: Error
+    tsh->>+Node: Connect
+    Node-->>+tsh: Success
+```
+
+Failed connection attempt when Auth is unavailable and per-session MFA is required:
+```mermaid
+sequenceDiagram
+    participant tsh
+    participant Auth
+    participant Node
+
+    tsh->>+Auth: IsMFARequired
+    Auth-->>-tsh: Error
+    tsh->>+Node: Connect
+    Node-->>+tsh: Access Denied
+```
+
+
+Alternative approaches:
+
+1) Attempt to connect to the node first, and if it fails with a `trace.AccessDenied` error then try `proto.AuthService/IsMFARequired` and then perform the MFA ceremony and connect to the node again
+    - Results in reduced latency for the happy path which doesn't require per-session MFA by removing the call to `proto.AuthService/IsMFARequired` entirely. However, `proto.AuthService/IsMFARequired` is still required   if the user doesn't have access to the node at all or per-session MFA is required.
+    - Results in increased latency when per-session MFA is required since that flow now requires establishing the connection to the target node twice.
+3) Perform `proto.AuthService/IsMFARequired` and connecting to the node in parallel and act accordingly
+    - Ultimately leads to roughly the same outcome as Option 1. The connection to the node almost always takes longer than the connection to auth. By performing them in parallel you really only get gains in the happy path.
+
+###### Session Trackers
+A new session tracker is created for every session. In the event that persisting the session tracker fails the 
+session is aborted. 
+
+Instead of immediately aborting the session if the session tracker cannot be persisted, the node could first check if 
+moderated sessions are required. In the event that they aren't the session should be allowed to proceed using a local 
+session tracker resource. 
+
+While this does allow access to the node, it will prevent the following:
+- the session from appearing in the active sessions list
+- the session will not be joinable by other users
+- the session recording will not be available
+
+
```

Comment:
Should we briefly discuss testing, so we don't relapse and start requiring Auth in all code paths again?